detection_rules/version_lock.py

```diff
@@ -82,10 +82,11 @@ def __getitem__(self, item) -> DeprecatedRulesEntry:
 
 def _convert_lock_version(stack_version: Optional[str]) -> Version:
```

Comment:
These changes add resiliency to allow rules to define M.m.p or M.m under `min_stack_version`, but always referencing (and saving) in the lock as M.m